### PR TITLE
All-around improvements to 0.18.0 docs

### PR DESCRIPTION
**Pull request by jtcohen6**
## Description & motivation

* Split out node selection syntax into several docs, with renames + redirects
* Add "best practice" for slim CI
* Net-new docs (additions to existing pages) for already merged, new-in-v0.18.0 features:
    * `full_refresh` node config
    * BigQuery: `policy_tags`, `hours_to_expiration`
    * Snowflake: `query_tag`
* Tiny reorg to docs on BQ service account impersonation

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [x] Yes: please update the base branch to `next`
- [ ] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [x] The page has been added to `website/sidebars.js`
- [x] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [x] The page has been removed from `website/sidebars.js`
- [x] An entry has been added to `_redirects`

### CODE REVIEW

File: website/docs/reference/node-selection/yaml-selectors.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+Selectors live in a top-level file named `selectors.yml`. Each has a `name` and a `definition`:
+
+<File name='selectors.yml'>
+
+```yml
+selectors:
+  - name: nodes_to_joy
+    definition: ...
+  - name: nodes_to_a_grecian_urn
```
Comment (drewbanin) [inline website/docs/reference/node-selection/yaml-selectors.md:27]: [community.getdbt.com/how-to-write-high-quality-documentation#using-obscure-greek-references](http://www.reactiongifs.com/r/jonw.gif)

Comment (jtcohen6) [reply website/docs/reference/node-selection/yaml-selectors.md:27]: [all you need to know](https://www.poetryfoundation.org/poems/44477/ode-on-a-grecian-urn)

File: website/docs/reference/node-selection/methods.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+```bash
+# These three selectors are equivalent
+dbt run --models package:snowplow
+dbt run --models snowplow
+dbt run --models snowplow.*
+```
+
+### The "config" method
+<Changelog>New in v0.18.0</Changelog>
+The `config` method is used to select models that match a specified [node config](config).
```
Comment (clrcrl) [inline website/docs/reference/node-selection/methods.md]: This is not rendering correctly:

```suggestion
<Changelog>New in v0.18.0</Changelog>

The `config` method is used to select models that match a specified [node config](config).
```

And so on and so forth through the rest of this page

**Review by clrcrl — CHANGES_REQUESTED** [review]
## Split out node selection syntax into several docs, with renames + redirects
Very supportive of the concept behind change! I think we can do even more here, e.g. I am a fan of how the test selection docs have very discrete example — one day we should get to that level for all node types.

A note to discuss: I think I've previously used the word "resource" where we are now using "node". We should aim to be consistent at some point, all the way down to making this consistent:
```
dbt ls --resource-type models
```

I also left a suggestion for a formatting error.

## The rest of the commits 
LGTM!

File: website/docs/reference/node-selection/methods.md (hunk 2 above)
Comment (jtcohen6) [reply website/docs/reference/node-selection/methods.md]: ooh nice catch! TIL

Comment (clrcrl) [reply website/docs/reference/node-selection/methods.md]: don't ask me why — i do not know

**Comment by jtcohen6** [comment]
Thanks for the review @clrcrl!

I realize that I've been using node and resource interchangeably, at some points using one as the definition of the other. ("Each dbt resource becomes a node in the DAG.") I agree that we should pick one, ultimately; for now I've just added a callout that, in fact, they mean the same thing.

**Review by drewbanin — APPROVED** [review]
This is all amazing - approved on my end :D

**Review by clrcrl — APPROVED** [review]
:shipit:

**Pull request merged**